### PR TITLE
feat: Calculate the cost of withdrawing MANA to Ethereum

### LINKED ISSUE #172: Calculate how much will cost the exit in Ethereum for MANA withdraws

### PR DESCRIPTION
**Pull request by juanmahidalgo**
Closes #172 

## Description

Introduces a new util method to calculate the cost of the withdrawal operation to Ethereum. 
Tech details:
- Uses the Gas Price API to get the current gas price (and uses the `fast` one). The `/estimates` endpoint also returns the current ETH price in USD, so it wasn't neccesarry to fetch it from Coingecko.
- The amount of gas consumed by the contract was gathered from a [withdrawal operation made in Goerli](https://goerli.etherscan.io/tx/0xc23b6ec66cd61f39245bd1b4fe0e5359526699f5159d7861a6c2bf19855544e8)

image
 

image

### CODE REVIEW

File: src/modules/mana/utils.ts [hunk 1]
```diff
@@ -25,15 +28,17 @@ import {
...
-export const MANA_CONTRACT_ADDRESS = process.env
-  .REACT_APP_MANA_CONTRACT_ADDRESS!
-export const ERC20_PREDICATE_CONTRACT_ADDRESS = process.env
-  .REACT_APP_ERC20_PREDICATE_CONTRACT_ADDRESS!
-export const ROOT_CHAIN_MANAGER_CONTRACT_ADDRESS = process.env
-  .REACT_APP_ROOT_CHAIN_MANAGER_CONTRACT_ADDRESS!
-export const MATIC_ROOT_CHAIN_SUBGRAPH = process.env
-  .REACT_APP_MATIC_ROOT_CHAIN_SUBGRAPH!
+export const MANA_CONTRACT_ADDRESS =
+  process.env.REACT_APP_MANA_CONTRACT_ADDRESS!
```
Comment (juanmahidalgo) [inline src/modules/mana/utils.ts:35]: Changes made by `prettier`

File: src/locales/en.json [hunk 2]
```diff
@@ -113,7 +113,11 @@
     "title_matic": "Convert to Ethereum MANA",
     "subtitle_matic": "Withdraw MANA from Polygon into Ethereum",
     "label_approvement": "MANA Approved",
-    "no_balance": "You don't have enough balance"
+    "no_balance": "You don't have enough balance",
+    "withdrawal_cost": "This operation consists of two steps. The first step won't have any cost. The second one will cost a approximately ${cost} of gas fees on Ethereum",
```
Comment (cazala) [inline src/locales/en.json:117]:
```suggestion
    "withdrawal_cost": "This operation consists of two steps. The first step won't have any cost. The second one will cost approximately {cost} of gas fees on Ethereum.",
```

File: src/components/Modals/ConvertManaModal/ConvertManaModal.tsx [hunk 3]
```diff
@@ -81,29 +110,27 @@ const ConvertManaModal: React.FC<Props> = ({
...
-              : 'convert_mana_modal.title_matic'
+  const content = useMemo(() => {
+    if (!hasAcceptedWithdrawalCost) {
+      return (
+        <div className="fees-warning">
+          {txEstimatedCost ? (
+            <T
+              id="convert_mana_modal.withdrawal_cost"
+              values={{
+                cost: <b>{txEstimatedCost.toFixed(2)}</b>,
```
Comment (cazala) [inline src/components/Modals/ConvertManaModal/ConvertManaModal.tsx:121]:
```suggestion
                cost: <b>${txEstimatedCost.toFixed(2)}</b>,
```

**Review by cazala — APPROVED** [review]
LGTM! I just left a suggestion to move the dollar sign within the bold tag

**Comment by juanmahidalgo** [comment]
If the API call fails, it automatically skips the step that shows the cost and renders the start withdrawal content.


https://user-images.githubusercontent.com/8763687/150997465-28067b2b-b5d7-4f2d-b84b-2bcdb45e9609.mov

**Review by cazala — APPROVED** [review]

**Pull request merged**